apache/trafficcontrol
# Every endpoint that modifies a CDN, or components related to a CDN, needs to be "lock safe"

## What does this PR (Pull Request) do?




## Which Traffic Control components are affected by this PR?


- Traffic Ops
- CI tests

## What is the best way to verify this PR?

Make sure all the unit tests and api tests pass.
Manual testing:
Run TO and TP locally.
Grab a hard lock for any cdn for a particular user.
Now, login to TP as another user, and try to modify the CDN, or CDN related components. These operations should be forbidden.
Log out and log back in to TP as the first user and try the same operations. This time they should all work correctly.

## If this is a bug fix, what versions of Traffic Control are affected?

- master

## The following criteria are ALL met by this PR


- [x] This PR does not include an update to CHANGELOG.md 


## Additional Information
